infrastructure/httpapi/handlers/crossfire.go

```diff
@@ -1,9 +1,162 @@
 package handlers
 
-/**
-Add crossfire HTTP Endpoint functions here
+import (
+	"fmt"
+	"strings"
 
-1. GetLatestRanks()
-2. GetParticipantByOperatorAddress()
+	"github.com/crypto-com/chain-indexing/appinterface/cosmosapp"
 
-**/
+	"github.com/valyala/fasthttp"
+
+	crossfire "github.com/crypto-com/chain-indexing/appinterface/projection/crossfire"
+	crossfire_constants "github.com/crypto-com/chain-indexing/appinterface/projection/crossfire/constants"
+	crossfire_validator_view "github.com/crypto-com/chain-indexing/appinterface/projection/crossfire/view"
+	"github.com/crypto-com/chain-indexing/appinterface/rdb"
+	"github.com/crypto-com/chain-indexing/infrastructure/httpapi"
+	applogger "github.com/crypto-com/chain-indexing/internal/logger"
+)
+
+type Crossfire struct {
+	logger applogger.Logger
+
+	validatorAddressPrefix string
+	consNodeAddressPrefix  string
+
+	cosmosAppClient              cosmosapp.Client
+	crossfireValidatorsView      *crossfire_validator_view.CrossfireValidators
+	crossfireValidatorsStatsView *crossfire_validator_view.CrossfireValidatorsStats
+	httpClient                   *crossfire.HTTPClient
+}
+
+func NewCrossfire(
+	logger applogger.Logger,
+	validatorAddressPrefix string,
+	consNodeAddressPrefix string,
+	cosmosAppClient cosmosapp.Client,
+	rdbHandle *rdb.Handle,
+	participantsListURL string,
+) *Crossfire {
+	return &Crossfire{
+		logger.WithFields(applogger.LogFields{
+			"module": "CrossfireHTTPHander",
+		}),
+
+		validatorAddressPrefix,
+		consNodeAddressPrefix,
+
+		cosmosAppClient,
+		crossfire_validator_view.NewCrossfireValidators(rdbHandle),
+		crossfire_validator_view.NewCrossfireValidatorsStats(rdbHandle),
+		crossfire.NewHTTPClient(participantsListURL),
+	}
+}
+
+// ListAllCrossfireValidators Fetch all validators and combine them
+func (handler *Crossfire) ListAllCrossfireValidators(ctx *fasthttp.RequestCtx) {
+	var err error
+
+	// Fetch Remote List
+	participantList, errFetching := handler.httpClient.Participants()
+	if errFetching != nil {
+		handler.logger.Errorf("error listing validators: %v", errFetching)
+		httpapi.InternalServerError(ctx)
+	}
+
+	// Fetch DB Crossfire Validators List
+	crossfireValidatorsDB, err := handler.crossfireValidatorsView.List()
+	if err != nil {
+		handler.logger.Errorf("error listing validators: %v", err)
+		httpapi.InternalServerError(ctx)
+		return
+	}
+
+	var finalList = []CrossfireValidatorDetails{}
+
+	for _, participant := range *participantList {
+		for _, dbValidator := range crossfireValidatorsDB {
+
+			// For each matching participant create a return object
+			if dbValidator.OperatorAddress == participant.OperatorAddress && dbValidator.InitialDelegatorAddress == participant.PrimaryAddress {
```

Comment:
For the `finalList`, so if there no `dbValidator.OperatorAddress` in the table, then it will return as empty array? As previous discussion, it's better to return the default value of the participants with the same structure.

As your response defined in the struct, should be like:
```
{
	operatorAddress.    FROM_PARTICIPANT
	initialDelegatorAddress    FROM_PARTICIPANT
	moniker                    FROM_PARTICIPANT
	taskSetup                  Incompleted
	taskKeepActive             Incompleted
	taskProposalVote           Incompleted
	taskNetworkUpgrade        Incompleted
	rankPhase1n2CommitmentRank 0
	rankPhase3CommitmentRank   0
	rankTxSentRank             0
	stats                     default all 0 ValidatorStats
}
```

Reply:
Yes. COrrect. You mentioned to keep it this way. I will push it soon.